src/reso/agent/ResoAgent.php

```diff
@@ -0,0 +1,186 @@
+    $param[] = $uploadId;
+    $this->dbManager->prepare($stmt, $sql);
+    $res = $this->dbManager->execute($stmt,$param);
+
+    return $this->dbManager->fetchAll($res);
+  }
+
+  /**
+   * @brief Find and associate base file to files containting license info
+   * @param array[] $reSoUploadFiles list of uploadtree objects to process
+   * @param string $uploadTreeTableName - uploadtree table for upload
+   * @return array[][] multi-dimensional array with license holding files with associated base files
+   */
+  protected function associateBaseFile($reSoUploadFiles, $uploadTreeTableName)
+  {
+    $mergedArray = array();
+    foreach ($reSoUploadFiles as $row) {
+      $baseFileRow = $this->findAssociatedFile($row, $uploadTreeTableName);
+      if (!empty($baseFileRow)) {
+        $row['assoc_file'] = $baseFileRow;
+        $mergedArray[] = $row;
+      }
+    }
+    return $mergedArray;
+  }
+
+  /**
+   * @brief Find associated base file.
+   * @param array[] $row - uploadtree entry to process
+   * @param string $uploadTreeTableName - uploadtree table for upload
+   * @return array[] uploadtree entry containing base file for input param
+   */
+  protected function findAssociatedFile($row, $uploadTreeTableName)
+  {
+    $stmt = __METHOD__ .'find_reso_base_file';
+    $sql = "SELECT * FROM $uploadTreeTableName
+      WHERE upload_fk=$1 AND ufile_name =$2 AND pfile_fk != 0 AND realparent=$3";
+    $param[] = $row['upload_fk'];
+    $param[] = substr($row['ufile_name'],0,-1 * abs(strlen(self::REUSE_FILE_SUFFIX)));
+    $param[] = $row['realparent'];
+    $this->dbManager->prepare($stmt, $sql);
+    $res = $this->dbManager->execute($stmt,$param);
+
+    return $this->dbManager->fetchAll($res);
+  }
+
+  /**
+   * @brief Cupy license from licene holder to base file
+   * @param array[][] $linkedFiles - multi-dimensional array with license holding files with associated base files
+   * @param int $uploadId
+   * @return true on success, Error on failure
+   */
+  protected function copyOjoFindings($linkedFiles,$uploadId)
+  {
+    //find agentId used for specific upload
+    $latestOjoAgent=$this->agentDao->agentARSList("ojo_ars",$uploadId);
+    $resoAgentId=$this->agentDao->getCurrentAgentId("reso");
+
+    foreach ($linkedFiles as $file) {
+      $this->heartbeat(1);
+      $param = array();
+      $insertParam = array();
+      $stmt = __METHOD__ .'readLicenseFindongsOjo';
+      $sql = "SELECT * FROM license_file
+        WHERE pfile_fk =$1 AND agent_fk=$2 and rf_fk IS NOT NULL";
+      $param[] = $file['pfile_fk'];
+      $param[] = $latestOjoAgent[0]['agent_fk'];
+
+      $this->dbManager->prepare($stmt, $sql);
+      $res = $this->dbManager->execute($stmt,$param);
+      while ($row=$this->dbManager->fetchArray($res)) {
+        $insertParam = array();
+        $Istmt = __METHOD__ .'insertLicenseFindongsReso';
+        $Isql = "INSERT INTO license_file (rf_fk, agent_fk, pfile_fk)
+          VALUES($1,$2,$3) RETURNING fl_pk";
```

Comment:
Please check if there already exists an entry with the same parameters before inserting (`rf_fk`, `agent_fk` and `pfile_fk`). Otherwise UI assumes there are multiple occurrences of the same license in the file and tries render the highlights (which does not exists) causing the bug mentioned in the description:

> Commas appear in the license table for files taht have been scanned multiple times. No incidence on workflow detected.
> ![image](https://user-images.githubusercontent.com/22956329/122936703-a51f7700-d371-11eb-8030-e1266c598556.png)

Reply:
Thanks for the solution!
I proposed a clean PR here: https://github.com/fossology/fossology/pull/2040
in which I do not insert if the combination is already in the table